theano/gof/graph.py

```diff
@@ -115,6 +116,13 @@ def __init__(self, op, inputs, outputs):
             else:
                 raise TypeError("The 'outputs' argument to Apply must contain Variable instances with no owner, not %s" % output)
 
+    def run_context(self):
+        """Returns the context for the node, or NoContext if no context is set.
+        """
+        if hasattr(self.op, 'get_context'):
+            return self.op.get_context(self)
```

Comment:
I didn't see the doc for the None context. What it mean? If no doc, add it.

On Wed, Jan 21, 2015 at 2:12 PM, abergeron <notifications@github.com> wrote:

> In theano/gof/graph.py
> <https://github.com/Theano/Theano/pull/2345#discussion_r23339237>:
>
> > @@ -115,6 +116,13 @@ def __init__(self, op, inputs, outputs):
> >              else:
> >                  raise TypeError("The 'outputs' argument to Apply must contain Variable instances with no owner, not %s" % output)
> >
> > +    def run_context(self):
> > +        """Returns the context for the node, or NoContext if no context is set.
> > +        """
> > +        if hasattr(self.op, 'get_context'):
> > +            return self.op.get_context(self)
>
> If there is no 'get_context' then I return NoContext to specify that there
> is no context.
>
> If get_context() returns None, then None is the context. That's a valid
> context value.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/2345/files#r23339237>.
>